reana_client/utils.py

```diff
@@ -173,6 +180,67 @@ def _prepare_kwargs(reana_yaml):
         raise e
 
 
+def _validate_environment(reana_yaml):
+    """Validate environments in REANA specification file according to workflow type.
+
+    :param reana_yaml: Dictionary which represents REANA specifications file.
+    """
+    workflow_type = reana_yaml["workflow"]["type"]
+
+    if workflow_type == "serial":
+        workflow_steps = reana_yaml["workflow"]["specification"]["steps"]
+        _validate_serial_workflow_environment(workflow_steps)
+    elif workflow_type == "yadage":
+        logging.warning(
+            "The environment validation is not implemented for Yadage workflows yet."
+        )
+    elif workflow_type == "cwl":
+        logging.warning(
+            "The environment validation is not implemented for CWL workflows yet."
+        )
+
+
+def _validate_serial_workflow_environment(workflow_steps):
+    """Validate environments in REANA serial workflow.
+
+    :param workflow_steps: List of dictionaries which represents different steps involved in workflow.
+    :raises Warning: Warns user if the workflow environment is invalid in serial workflow steps.
+    """
+    for step in workflow_steps:
+        if step.get("environment", None):
```

Comment:
It should check if it has no tag, instead of no environment. For instance, check `reana-demo-lhcb-d2pimumu` example that has the environment [`reanahub/reana-env-root6`](https://github.com/reanahub/reana-demo-lhcb-d2pimumu/blob/7c008deb1974f25080a663711f530a0a515dda4f/reana.yaml#L16) (no tag included). In this case it should raise the warning.

Reply:
Fixed in https://github.com/reanahub/reana-client/pull/459/commits/0b466c29bd6aadefba8a8afdd98a931a76b7cbaf